codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -0,0 +1,642 @@
+                     if !response.status().is_success() && response.status().as_u16() != $successCode {
+                        #{parse_error}(response)
+                     } else {
+                        #{parse_response}(response)
+                     }
+                }
+            }""",
+            *codegenScope,
+            "O" to outputSymbol,
+            "E" to operationShape.errorSymbol(symbolProvider),
+            "parse_error" to httpBoundProtocolGenerator.parseError(operationShape),
+            "parse_response" to httpBoundProtocolGenerator.parseResponse(operationShape)
+        )
+        /* Implement ParseHttpRequest for non streaming types. This is done by only implementing `parse_loaded` */
+        rustTemplate(
+            """
+            impl #{ParseHttpRequest} for $operationName {
+                type Input = std::result::Result<#{I}, #{Error}>;
+                fn parse_unloaded(&self, _request: &mut #{Http}::Request<#{SdkBody}>) -> Option<Self::Input> {
+                    None
+                }
+                fn parse_loaded(&self, request: &#{Http}::Request<#{Bytes}>) -> Self::Input {
+                    #{parse_request}(request)
+                }
+            }""",
+            *codegenScope,
+            "I" to inputSymbol,
+            "parse_request" to serverParseRequest(operationShape)
+        )
+        /* Implement SerializeHttpResponse for non streaming types. This is done by only implementing `serialize` */
+        rustTemplate(
+            """
+            impl #{SerializeHttpResponse} for $operationName {
+                type Output = std::result::Result<#{Http}::Response<#{Bytes}>, #{Error}>;
+                type Struct = #{O};
+                fn serialize(&self, output: &Self::Struct) -> Self::Output {
+                    #{serialize_response}(output)
+                }
+            }""",
+            *codegenScope,
+            "O" to outputSymbol,
+            "serialize_response" to serverSerializeResponse(operationShape)
+        )
+        /* Implement SerializeHttpError for non streaming types. This is done by only implementing `serialize` */
+        rustTemplate(
+            """
+            impl #{SerializeHttpError} for $operationName {
+                type Output = std::result::Result<#{Http}::Response<#{Bytes}>, #{Error}>;
+                type Struct = #{E}Kind;
+                fn serialize(&self, error: &Self::Struct) -> Self::Output {
+                    #{serialize_error}(error)
+                }
+            }""",
+            *codegenScope,
+            "E" to errorSymbol,
+            "serialize_error" to serverSerializeError(operationShape)
+        )
+    }
+
+    /*
+     * TODO: implement straming traits
+     */
+    private fun RustWriter.renderStreamingTraits(
+        operationName: String,
+        outputSymbol: Symbol,
+        operationShape: OperationShape
+    ) {
+        logger.warning(
```

Comment:
_generally_ I find that it's useful to generate code with `todo!("some message") /* generated comment indicating what the problem was */` in it, although I guess in this case, that doesn't quite work.

Reply:
In this specific case I don't think I can put a todo! inside rust as it will be not inside a function. This is the top level definition of the streaming type trait.